components/camel-consul/src/main/java/org/apache/camel/component/consul/cloud/ConsulServiceDiscovery.java

```diff
@@ -62,7 +62,12 @@ public ConsulServiceDiscovery(ConsulConfiguration configuration) throws Exceptio
             .getResponse();
 
         return services.stream()
-            .map(service -> newService(name, service, healths))
+            .map(service -> newService(
+                    name,
+                    service,
+                    healths.stream()
+                            .filter(serviceHealth -> serviceHealth.getService().getId().equals(service.getServiceId()))
+                            .collect(Collectors.toList())))
```

Comment:
Wondering if this can be moved inside `newService`

Reply:
Done!